### PR TITLE
[Refactor:Controller&Router] Separate GradeInquiryController from SubmissionController

### PR DESCRIPTION
**Pull request by zjxiaohan**
### Please check if the PR fulfills these requirements:

* [x] The PR title and message follows our [guidelines](https://submitty.org/developer/how_to_contribute)

### What is the current behavior?
SubmissionController is such a monster that I feel very stressful editing it.

### What is the new behavior?
Separate code regarding regrading from SubmissionController, and add some routes.

### Other information?
New URLs:
- `/{_semester}/{_course}/gradeable/{gradeable_id}/regrade/new` POST ONLY
- `/{_semester}/{_course}/gradeable/{gradeable_id}/regrade/post` POST ONLY
- `/{_semester}/{_course}/gradeable/{gradeable_id}/regrade/toggle_status` POST ONLY

Also changed the URL for a single gradeable from `/{_semester}/{_course}/student/{gradeable_id}` to `/{_semester}/{_course}/gradeable/{gradeable_id}`.

Manually tested:
- Opening a grade inquiry
- Close it and reopen it from instructor's side and from student's side
- Make posts as a student and as an instructor

### CODE REVIEW

**Comment by zjxiaohan** [comment]
So I check the course settings when creating an inquiry by calling `isRegradeOpen` (which checks course settings behind the scenes)

https://github.com/Submitty/Submitty/blob/5e642a0ea12428b7fb54b5991409cfc593442ccc/site/app/controllers/student/RegradeController.php#L34

and for other functions, I think it suffices to have these lines:

https://github.com/Submitty/Submitty/blob/5e642a0ea12428b7fb54b5991409cfc593442ccc/site/app/controllers/student/RegradeController.php#L90

https://github.com/Submitty/Submitty/blob/5e642a0ea12428b7fb54b5991409cfc593442ccc/site/app/controllers/student/RegradeController.php#L142

File: site/app/controllers/student/RegradeController.php [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+
+use app\controllers\AbstractController;
+use app\models\Notification;
+use app\models\Email;
+use app\libraries\response\Response;
+use app\libraries\response\JsonResponse;
+use Symfony\Component\Routing\Annotation\Route;
+
+
+class RegradeController extends AbstractController {
```
Comment (sak6lab) [inline site/app/controllers/student/RegradeController.php]: Rename to GradeInquiryController

File: site/app/controllers/student/RegradeController.php [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+    public function run() {
+        return null;
+    }
+
+    /**
+     * @param $gradeable_id
+     * @Route("/{_semester}/{_course}/gradeable/{gradeable_id}/regrade/new", methods={"POST"})
+     * @return Response|null null is for tryGetGradeable and tryGetGradedGradeable
+     */
+    public function requestRegrade($gradeable_id) {
```
Comment (sak6lab) [inline site/app/controllers/student/RegradeController.php]: rename to requestGradeInquiry

File: site/app/controllers/student/RegradeController.php [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+            );
+        }
+    }
+
+    /**
+     * @param $gradeable_id
+     * @Route("/{_semester}/{_course}/gradeable/{gradeable_id}/regrade/post", methods={"POST"})
+     * @return Response|null null is for tryGetGradeable and tryGetGradedGradeable
+     */
+    public function makeRequestPost($gradeable_id) {
```
Comment (sak6lab) [inline site/app/controllers/student/RegradeController.php]: rename to makeGradeInquiryPost

File: site/app/controllers/student/RegradeController.php [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+            );
+        }
+    }
+
+    /**
+     * @param $gradeable_id
+     * @Route("/{_semester}/{_course}/gradeable/{gradeable_id}/regrade/toggle_status", methods={"POST"})
+     * @return Response|null null is for tryGetGradeable and tryGetGradedGradeable
+     */
+    public function changeRequestStatus($gradeable_id) {
```
Comment (sak6lab) [inline site/app/controllers/student/RegradeController.php]: rename to changeGradeInquiryStatus

File: site/app/controllers/student/RegradeController.php [hunk 5]
```diff
@@ -0,0 +1,250 @@
...
+
+
+class RegradeController extends AbstractController {
+    public function run() {
+        return null;
+    }
+
+    /**
+     * @param $gradeable_id
+     * @Route("/{_semester}/{_course}/gradeable/{gradeable_id}/regrade/new", methods={"POST"})
```
Comment (sak6lab) [inline site/app/controllers/student/RegradeController.php]: replace regrade in routes to grade_inquiry

**Comment by zjxiaohan** [comment]
oops I forgot to change the URL of the buttons. Sorry!!

**Pull request merged**